packages/world/ts/internal/genInitRecipesSol.ts

```diff
@@ -0,0 +1,52 @@
+import type { ObjectTypeName } from "../objects";
+import { type Recipe, recipes } from "../recipes";
+
+function renderRecipe(recipe: Recipe): string {
```

Comment:
We should do the check here to ensure that the mass & energy of outputs is as expected, ie conserved

Reply:
ahh I see `validateRecipeMassEnergy` but doesnt seem like its getting called anywhere?

Reply:
yeah wasn't using it, now I'm validating on gen, but currently only emitting a warning due to the furnace recipes (they have coal input which shouldn't be counted as part of the recipe)